### PR TITLE
MAC: add MAC_TX_QUEUE_FULL error code, and account it in link stats separately

### PR DESCRIPTION
**Pull request by atiselsts**
Allows to collect statistics on queue losses, which are important for network performance evaluation.

### CODE REVIEW

**Comment by atiselsts** [comment]
In order to see a demonstration of this functionality, you can check the `examples/benchmarks/result-visualization` example - it has `LINK_STATS_CONF_PACKET_COUNTERS` configuration enabled, so it now collects statistics about the queue losses and prints them. `run-cooja.py` followed by `run-analysis.py` should print the queue losses (among other stats).

File: os/net/link-stats.c [hunk 1]
```diff
@@ -166,6 +166,14 @@ link_stats_packet_sent(const linkaddr_t *lladdr, int status, int numtx)
   stats->last_tx_time = clock_time();
   stats->freshness = MIN(stats->freshness + numtx, FRESHNESS_MAX);
 
+  if(status == MAC_TX_QUEUE_FULL) {
```
Comment (nfi) [inline os/net/link-stats.c:169]: Full queues were earlier reported as `MAC_TX_ERR` which did not update last timestamp and freshness (`stats->last_tx_time` and `stats->freshness`). What is the motivation of changing this behaviour?

Comment (atiselsts) [reply os/net/link-stats.c:169]: A good point - this PR is not meant to change that logic. Don't remember exactly why it was made this way, probably an oversight on my part.

Moving this `if` statement to line 166 should be sufficient to get the original behavior, right?

Comment (atiselsts) [reply os/net/link-stats.c:165]: changed the code by moving the `if` block before the freshness update logic

**Review by nfi — APPROVED** [review]

**Comment by joakimeriksson** [comment]
We need to remember to add this to the change-list.

**Pull request merged**